File: src-c++/adjacent.cpp
#include "adjacent.h"

Adjacent::Adjacent(const Triangle &triangle) : triangle(triangle) {
    mapAngle = 0.0;
    otherCoord = std::make_pair(NULL, NULL);
    basealtiX = std::make_pair(NULL, NULL);
    basealtiH = std::make_pair(NULL, NULL);
    mappedHeight = NULL;
}

std::string Adjacent::toString() const {
    return "Triangle(" + (triangle.ID) +
           "): base-coord=(" + std::to_string(otherCoord.first) + "," + std::to_string(otherCoord.second) +
           "), map angle= " + std::to_string(mapAngle) +
           " | xpos altitude=(" + std::to_string(basealtiX.first) + "," + std::to_string(basealtiX.second) +
           ") | altitude height=(" + std::to_string(basealtiH.first) + "," + std::to_string(basealtiH.second) +
           ")-> Mapped Height = " + std::to_string(mappedHeight);
}

void Adjacent::setCoord(std::pair<int, double> coord) {
    otherCoord = coord;
}

void Adjacent::setAngle(double angle) {
    mapAngle = angle;
}

void Adjacent::setAltiX(std::pair<int, double> xpos) {
    basealtiX = xpos;
}

void Adjacent::setAltiH(std::pair<int, double> height) {
    basealtiH = height;
}

void Adjacent::setMapHeight(std::pair<double, double, double> height) {
    mappedHeight = height;
}

File: src/adjacent.py
from triangle import Triangle

class Adjacent:

    def __init__(self, triangle: Triangle):
        self.triangle = triangle
        self.mapAngle = 0
        self.otherCoord = (None,0)
        self.basealtiX = None
        self.basealtiH = None
        self.mappedHeight = None

    def __str__(self) -> str:
        return f"Triangle({self.triangle.ID}): base-coord=({self.otherCoord[0]},{self.otherCoord[1]:.3f}), map angle= {self.mapAngle:.3f} | xpos altitude=({self.basealtiX[0]},{self.basealtiX[1]:.3f}) | altitude height=({self.basealtiH[0]},{self.basealtiH[1]:.3f})-> Mapped Height = {self.mappedHeight}"
    
    def __repr__(self):
        return str(self)
    
    def setCoord(self, coord):
        self.otherCoord = coord
    
    def setAngle(self, angle):
        self.mapAngle = angle

    def setAltiX(self, xpos):
        self.basealtiX = xpos
    
    def setAltiH(self, height):
        self.basealtiH = height

    def setMapHeight(self, height):
        self.mappedHeight = height